src/platform/native.zig

```diff
@@ -173,6 +175,18 @@ pub const Platform = struct {
         engine.current_desc = descriptor;
         engine.target_desc = descriptor;
 
+        var mouse_cursors: [@typeInfo(enums.MouseCursor).Enum.fields.len]glfw.Cursor = undefined;
+        mouse_cursors[@enumToInt(enums.MouseCursor.arrow)] = try glfw.Cursor.createStandard(.arrow);
+        mouse_cursors[@enumToInt(enums.MouseCursor.ibeam)] = try glfw.Cursor.createStandard(.ibeam);
+        mouse_cursors[@enumToInt(enums.MouseCursor.crosshair)] = try glfw.Cursor.createStandard(.crosshair);
+        mouse_cursors[@enumToInt(enums.MouseCursor.pointing_hand)] = try glfw.Cursor.createStandard(.pointing_hand);
+        mouse_cursors[@enumToInt(enums.MouseCursor.resize_ew)] = try glfw.Cursor.createStandard(.resize_ew);
+        mouse_cursors[@enumToInt(enums.MouseCursor.resize_ns)] = try glfw.Cursor.createStandard(.resize_ns);
+        mouse_cursors[@enumToInt(enums.MouseCursor.resize_nwse)] = try glfw.Cursor.createStandard(.resize_nwse);
+        mouse_cursors[@enumToInt(enums.MouseCursor.resize_nesw)] = try glfw.Cursor.createStandard(.resize_nesw);
+        mouse_cursors[@enumToInt(enums.MouseCursor.resize_all)] = try glfw.Cursor.createStandard(.resize_all);
+        mouse_cursors[@enumToInt(enums.MouseCursor.not_allowed)] = try glfw.Cursor.createStandard(.not_allowed);
```

Comment:
Not all of these cursors are available on all platforms with GLFW, and so all these `try` statements will actually make `init` just outright fail on some systems (primarily Linux and macOS)

What we'll need to do is handle cursor requests in a similar way to GLFW, calling `createStandard` only when needed and storing the result, and properly returning an error in the event the cursor is not available.

Reply:
Alternatively, we could say for now that if setting the cursor fails we make it a no-op (ignore the error or log it), and in the future produce custom cursor images with GLFW for any OS that does not have native cursors for these.

I kind of like that idea a little better, because then it reduces the chances someone does **exactly this**: think they're available everywhere, and write an application that just panics/crashes on linux/macos by using `try`.

Thoughts?